app/controllers/authentication_controller.rb

```diff
@@ -34,6 +47,15 @@ def trigger_jobseeker_sign_in_event(success_or_failure, errors = nil)
   end
 
   def trigger_successful_publisher_sign_in_event(sign_in_type, publisher_oid = nil)
+    event = DfE::Analytics::Event.new
+      .with_type(:successful_publisher_sign_in_attempt)
+      .with_request_details(request)
+      .with_response_details(response)
+      .with_user(current_publisher)
+      .with_data(user_anonymised_publisher_id: StringAnonymiser.new(publisher_oid), sign_in_type: sign_in_type)
```

Comment:
Not sure if adding `user_anonymised_publisher_id` is required, as the `with_user` method can now be configured to anonymise the user identifier. Please see: https://github.com/DFE-Digital/dfe-analytics/pull/64

Also we must use either `id` or `oid` consistently for any given entity. For example, if the `analytics.yml` and `analytics_pii.yml` specify `publisher_id` the in the user_id field we must use `publisher.id`.


This is a general comment that is applicable throughout this PR.